src/main/java/com/powsybl/caseserver/service/CaseService.java

```diff
@@ -0,0 +1,170 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.caseserver.service;
+
+import com.powsybl.caseserver.CaseException;
+import com.powsybl.caseserver.dto.CaseInfos;
+import com.powsybl.caseserver.dto.ExportCaseInfos;
+import com.powsybl.caseserver.parsers.FileNameInfos;
+import com.powsybl.caseserver.parsers.FileNameParser;
+import com.powsybl.caseserver.parsers.FileNameParsers;
+import com.powsybl.caseserver.repository.CaseMetadataEntity;
+import com.powsybl.caseserver.repository.CaseMetadataRepository;
+import com.powsybl.commons.datasource.DataSource;
+import com.powsybl.commons.datasource.DataSourceUtil;
+import com.powsybl.commons.datasource.MemDataSource;
+import com.powsybl.computation.ComputationManager;
+import com.powsybl.iidm.network.Exporter;
+import com.powsybl.iidm.network.Importer;
+import com.powsybl.iidm.network.Network;
+import org.springframework.web.multipart.MultipartFile;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+public interface CaseService {
+    default void validateCaseName(String caseName) {
+        Objects.requireNonNull(caseName);
+        if (!caseName.matches("[^<>:\"/|?*]+(\\.[\\w]+)")) {
+            throw CaseException.createIllegalCaseName(caseName);
+        }
+    }
+
+    default CaseInfos createInfos(String fileBaseName, UUID caseUuid, String format) {
+        FileNameParser parser = FileNameParsers.findParser(fileBaseName);
+        if (parser != null) {
+            Optional<? extends FileNameInfos> fileNameInfos = parser.parse(fileBaseName);
+            if (fileNameInfos.isPresent()) {
+                return CaseInfos.create(fileBaseName, caseUuid, format, fileNameInfos.get());
+            }
+        }
+        return CaseInfos.builder().name(fileBaseName).uuid(caseUuid).format(format).build();
+    }
+
+    default void createCaseMetadataEntity(UUID newCaseUuid, boolean withExpiration, boolean withIndexation, String originalFilename, String compressionFormat, CaseMetadataRepository caseMetadataRepository) {
```

Comment:
add getCaseMetadataRepository so that code here in and other methods can call it instead of passing the caseMetadataRepository as an argument

Reply:
done